recipes/arrow/all/conanfile.py

```diff
@@ -155,39 +157,55 @@ def config_options(self):
         if Version(self.version) < "8.0.0":
             del self.options.substrait
 
+        self.options.parquet = self._parquet()
+        self.options.compute = self._compute()
+        self.options.dataset_modules = self._dataset_modules()
+        self.options.with_boost = self._with_boost()
+        self.options.with_flight_rpc = self._with_flight_rpc()
+        self.options.with_gflags = self._with_gflags()
+        self.options.with_glog = self._with_glog()
+        self.options.with_grpc = self._with_grpc()
+        self.options.with_jemalloc = self._with_jemalloc()
+        self.options.with_thrift = self._with_thrift()
+        self.options.with_llvm = self._with_llvm()
+        self.options.with_openssl = self._with_openssl()
+        self.options.with_protobuf = self._with_protobuf()
+        self.options.with_re2 = self._with_re2()
+        self.options.with_utf8proc = self._with_utf8proc()
```

Comment:
I just want to mention that all this logic is broken, since all these functions try to access options values which don't even exist at `config_options()` time.

Reply:
what is the proper way to define dependencies between options in conan 2?